### PR TITLE
MKL related files with review comments incorporated

### PR DESCRIPTION
**Pull request by rgomathi**
This PR is based on commit "977c6b3" as this version allows MKL to use all the cores available.
All MKL related files are added here after incorporating review comments, major changes include

1. usage of Clang-format(Linter) with --style = Google
2. usage of macros for checking input and filter dimension in the mkl operators
3. merged Max and Average pooling functions
4. created a new folder for mkl related python scripts in Python folder and moved them there
5. there is no mkl_alexnet_test.py as that was redundant while convnet_benchmark.py does the same thing

### CODE REVIEW

File: caffe2/python/core.py [hunk 1]
```diff
@@ -1675,6 +1675,11 @@ def RunAllOnGPU(self, gpu_id=0, use_cudnn=False):
         if use_cudnn:
             for op in self._net.op:
                 op.engine = "CUDNN"
+    def RunAllOnMKL(self):
+        """A convenient function to run everything on the GPU."""
```
Comment (salexspb) [inline caffe2/python/core.py:1679]: nit: GPU -> MKL

**Comment by salexspb** [comment]
Looks like there are build errors when I run this internally. Attaching a gist with full logs. 
https://gist.github.com/salexspb/4a39645c04572e09ba76cd5e96f40136

**Comment by Yangqing** [comment]
@githubGomathi it seems to be that our internal MKL is still at 2017.0.098. I am updating it overnight and after that, we should be able to verify and merge the PR. In the meantime feel free to do benchmarks on your local branch.

**Comment by rgomathi** [comment]
sure... thanks Yangqing !

**Comment by salexspb** [comment]
There are a lot of compilation errors, seems that part of the code is not exercised. I have sent more details over e-mail.

**Pull request closed without merging**